packages/cloud_firestore/ios/Classes/CloudFirestorePlugin.m

```diff
@@ -419,26 +496,21 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
     }];
   } else if ([@"Query#addSnapshotListener" isEqualToString:call.method]) {
     __block NSNumber *handle = [NSNumber numberWithInt:_nextListenerHandle++];
-    FIRQuery *query;
-    @try {
-      query = getQuery(call.arguments);
-    } @catch (NSException *exception) {
-      result([FlutterError errorWithCode:@"invalid_query"
-                                 message:[exception name]
-                                 details:[exception reason]]);
-    }
-    id<FIRListenerRegistration> listener = [query
-        addSnapshotListener:^(FIRQuerySnapshot *_Nullable snapshot, NSError *_Nullable error) {
-          if (snapshot == nil) {
-            result(getFlutterError(error));
-            return;
-          }
-          NSMutableDictionary *arguments = [parseQuerySnapshot(snapshot) mutableCopy];
-          [arguments setObject:handle forKey:@"handle"];
-          [self.channel invokeMethod:@"QuerySnapshot" arguments:arguments];
-        }];
-    _listeners[handle] = listener;
-    result(handle);
+    getQuery(call.arguments, ^(FIRQuery *query) {
+      id<FIRListenerRegistration> listener = [query
+          addSnapshotListener:^(FIRQuerySnapshot *_Nullable snapshot, NSError *_Nullable error) {
+            if (snapshot == nil) {
+              result(getFlutterError(error));
+              return;
+            }
+            NSMutableDictionary *arguments = [parseQuerySnapshot(snapshot) mutableCopy];
+            [arguments setObject:handle forKey:@"handle"];
+            [self.channel invokeMethod:@"QuerySnapshot" arguments:arguments];
```

Comment:
Using `self` in this block seems can cause retain cycle? Maybe use weakSelf instead?